mopidy/models/__init__.py

```diff
@@ -253,6 +253,9 @@ class Track(ValidatedImmutableObject):
     #: equivalent timestamp or simply a version counter.
     last_modified = fields.Integer(min=0)
 
+    #: The track source, e.g. "spotify" or "local". Read-only.
+    source = fields.String()
```

Comment:
This could also be the Identifier type, which would limit the values to `str` and then also intern the values meaning we can represent each value with a pointer instead of the full value. Which should save some memory given our inefficient model which isn't de-normalized and that some users have quite large music collections.